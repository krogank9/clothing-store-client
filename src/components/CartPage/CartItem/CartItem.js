import React from 'react';
import { Link } from "react-router-dom";

import ClothingStoreContext from '../../../contexts/ClothingStoreContext.js';

import Utils from "../../../utils";

class CartItem extends React.Component {
  static contextType = ClothingStoreContext;

  static defaultProps = {
    product: {},
    quantity: 0
  }

  constructor() {
    super();

    this.state = {
    }
  }

  onRemoveClicked = () => {
    this.context.removeFromCart(this.props.product.id)
  }

  getSize = () => {
    //this.props.size
    return false//"Medium"
  }

  getProductLink = () => {
    return `/products/${Utils.normalizeName(this.props.product.name)}.${this.props.product.id}`
  }

  getPrice = () => {
    return Utils.normalizePrice(this.props.product.price)
  }

  getQuantityPrice = () => {
    return Utils.normalizePrice(this.props.product.price * this.props.quantity)
  }

  render() {
    return (
      <tr>
        <td>
          <div className="cart-item">
            <div className="cart-item-img-container">
              <Link to={this.getProductLink()}>
                <img className="cart-product-img" src={`${process.env.PUBLIC_URL}/assets/products/${Utils.normalizeName(this.props.product.name)}.jpg`}></img>
              </Link>
            </div>
            <div className="cart-item-info padding-15">
              <div><Link to={this.getProductLink()}>{this.props.product.name}</Link></div>
              {this.getPrice()}
              <div className="mobile-only">
                Qty: {this.props.quantity}
                <br />
                <Link to="#" onClick={this.onRemoveClicked}>Remove</Link>
              </div>
            </div>
          </div>
        </td>
        <td className="desktop-only">
          <div className="padding-5">
            {this.props.quantity}
            <br />
            <Link to="#" onClick={this.onRemoveClicked}>Remove</Link>
          </div>
        </td>
        <td className="desktop-only">
          <div className="padding-5">
            ${this.getQuantityPrice()}
          </div>
        </td>
      </tr>
    );
  }
}

export default CartItem;