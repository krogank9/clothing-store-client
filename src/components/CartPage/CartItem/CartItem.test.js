import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from "react-router-dom";
import CartItem from './CartItem';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`CartItem component`, () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <table><tbody>
          <CartItem />
        </tbody></table>
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('renders the App', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <table><tbody>
          <CartItem />
        </tbody></table>
      </BrowserRouter>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})