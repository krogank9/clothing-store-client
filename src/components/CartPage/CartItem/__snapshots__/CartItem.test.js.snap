// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component renders the App 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <table>
    <tbody>
      <CartItem
        product={Object {}}
        quantity={0}
      />
    </tbody>
  </table>
</Router>
`;
